File: Java/백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class Main {
    private static final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());
        Deque<Integer> queue = new LinkedList<>();
        while(N-- > 0){
            StringTokenizer st = new StringTokenizer(br.readLine());
            switch(st.nextToken()){
                case "push" -> queue.offer(Integer.parseInt(st.nextToken()));
                case "pop" -> sb.append(queue.isEmpty() ? -1 : queue.poll()).append("\n");
                case "size" -> sb.append(queue.size()).append("\n");
                case "empty" -> sb.append(queue.isEmpty() ? 1 : 0).append("\n");
                case "front" -> sb.append(queue.isEmpty() ? -1 : queue.peek()).append("\n");
                case "back" -> sb.append(queue.isEmpty() ? -1 : queue.peekLast()).append("\n");
            }
        }
        System.out.println(sb);
    }
}

File: Kotlin (JVM)/백준/Silver/18258. 큐 2/큐 2.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>){
    var sb: StringBuilder = StringBuilder()
    var scan: BufferedReader = BufferedReader(InputStreamReader(System.`in`))
    var N: Int = scan.readLine().toInt()
    var queue: Deque<Int> = LinkedList()
    while (N-- > 0){
        var st: StringTokenizer = StringTokenizer(scan.readLine())
        when(st.nextToken()){
            "push" -> { queue.offer(st.nextToken().toInt()) }
            "pop" -> { sb.append(if(queue.isEmpty()) -1 else queue.poll()).append("\n")}
            "size" -> {sb.append(queue.size).append("\n")}
            "empty" -> {sb.append(if(queue.isEmpty()) 1 else 0).append("\n")}
            "front" -> {sb.append(if(queue.isEmpty()) -1 else queue.peek()).append("\n")}
            "back" -> {sb.append(if(queue.isEmpty()) -1 else queue.peekLast()).append("\n")}
        }
    }
    println(sb)
}